src/decoders/mod.rs

```diff
@@ -182,15 +184,94 @@ impl Camera {
       bps: 0,
       wb_offset: 0,
       hints: Vec::new(),
+      orientation: Orientation::Unknown,
     }
   }
 }
 
-pub fn ok_image(camera: &Camera, width: usize, height: usize, wb_coeffs: [f32;4], image: Vec<u16>) -> Result<RawImage,String> {
+/// Image operations to perform. Instances of this struct should normally be
+/// constructed using `Orientation::into()`
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub struct OrientationFlips {
+  pub transpose: bool,
+  pub flip_x: bool,
+  pub flip_y: bool,
+}
+
+/// Possible values for the IFD tag Orientation (0x0112)
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub enum Orientation {
+  Normal,
+  HorizontalFlip,
+  Rotate180,
+  VerticalFlip,
+  Transpose,
+  Rotate90,
+  Transverse,
+  Rotate270,
+  Unknown,
+}
+
+impl Orientation {
+  /// Convert a u16 into its corresponding enum value
+  pub fn from_u16(orientation: u16) -> Orientation {
+    match orientation {
+      1 => Orientation::Normal,
+      2 => Orientation::HorizontalFlip,
+      3 => Orientation::Rotate180,
+      4 => Orientation::VerticalFlip,
+      5 => Orientation::Transpose,
+      6 => Orientation::Rotate90,
+      7 => Orientation::Transverse,
+      8 => Orientation::Rotate270,
+      _ => Orientation::Unknown,
+    }
+  }
+
+  /// Extract orientation from given EXIF data
+  pub fn from_tiff_ifd(tiff: &TiffIFD) -> Orientation {
+    // Lookup the orientation of the image to store for rotating after demosaic
+    if let Some(ifd) = tiff.find_first_ifd(Tag::Orientation) {
+      // Since we got here it should always be safe to unwrap
+      Orientation::from_u16(ifd.find_entry(Tag::Orientation).unwrap().get_usize(0) as u16)
+    } else {
+      Orientation::Unknown
+    }
+  }
+}
+impl Into<OrientationFlips> for Orientation {
+  fn into(self) -> OrientationFlips {
+    OrientationFlips {
+      transpose: match self {
```

Comment:
Wouldn't it be simpler to just do a single match on all the orientation flips and do OrientationFlips{bool,bool,bool} for each of them?